File: C/Algorithms/53.c
int maxSubArray(int *nums, int numsSize) {
  int maximum = nums[0], prev = 0;
  for (int i = 0; i < numsSize; i++) {
    int n = *(nums + i);
    int k = n + prev;
    prev = n > k ? n : k;
    maximum = prev > maximum ? prev : maximum;
  }
  return maximum;
}

File: Rust/Algorithms/53.rs
use std::cmp::max;

impl Solution {
    pub fn max_sub_array(nums: Vec<i32>) -> i32 {
        let mut maximum = nums[0];
        let mut prev = 0;
        for num in nums {
            prev = max(num, prev + num);
            maximum = max(maximum, prev);
        }
        maximum
    }
}
